### PR TITLE
Analytics base client

### PR DESCRIPTION
**Pull request by simons5593**
```
session = Session(account="lifeomic", token=token)
client = Analytics(session)
search = SubjectSearchQueryBuilder()
search.patient() \
    .observations() \
    .code(eq='11142-7') \
    .system(eq='http://loinc.org') \
    .value_quantity(eq=18)


res = client.get_patients(project='5a07dedb-fa2a-4cb0-b662-95b23a050221', query_builder=search)

```

### CODE REVIEW

File: phc/web/analytics.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import json
+
+from phc.web.base_client import BaseClient
+
+
+class Analytics(BaseClient):
+    """Provides analytics subject search"""
+
+    def jprint(self, data):
```
Comment (mschroering) [inline phc/web/analytics.py]: Think this should live on the `ApiResponse`? Then we could pretty print any response.

File: tests/unit/tests.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import unittest
```
Comment (mschroering) [inline tests/unit/tests.py]: If you look at `master`, i was having a hard time getting these tests to run, so I moved them to be under `phc/tests/subject_search_test.py` to make the importing work.

Comment (simons5593) [reply tests/unit/tests.py]: Whoops, I must have resolved the conflict incorrectly. Will take a look.

**Comment by mjtieman** [comment]
[Warning: naming suggestion]: it looks like we want a pattern of "pre-packaging" the `domain` and `target` properties of the DSL into classes, e.g. `SubjectSearchQueryBuilder` is focused on `patient` target and `filter` domain.

Should we name these query builds to reflect that, e.g. `PatientFilterQueryBuilder`? When we get to variants, it would be `VariantFilterQueryBuilder`.

**Comment by simons5593** [comment]
> [Warning: naming suggestion]: it looks like we want a pattern of "pre-packaging" the `domain` and `target` properties of the DSL into classes, e.g. `SubjectSearchQueryBuilder` is focused on `patient` target and `filter` domain.
> 
> Should we name these query builds to reflect that, e.g. `PatientFilterQueryBuilder`? When we get to variants, it would be `VariantFilterQueryBuilder`.

Sure, will rename

**Pull request merged**